src/utils/nftRewards.ts

```diff
@@ -320,14 +320,33 @@ export function decodeJB721DelegateRedeemMetadata(
   }
 }
 
+// returns an array of NftRewardTiers corresponding to a given list of tier IDs
+//    Note: If ids contains multiple of the same ID, the return value
+//          will contain corresponding rewardTier multiple times.
+export function rewardTiersFromIds({
+  ids,
+  rewardTiers,
+}: {
+  ids: number[]
+  rewardTiers: NftRewardTier[]
+}) {
+  return ids
+    .map(id => rewardTiers.find(tier => tier.id === id))
+    .filter(tier => Boolean(tier)) as NftRewardTier[]
+}
+
 // sums the contribution floors of a given list of nftRewardTiers
 //    - optional select only an array of ids
 export function sumTierFloors(rewardTiers: NftRewardTier[], ids?: number[]) {
-  if (ids) {
-    rewardTiers = rewardTiers.filter(tier => ids.includes(tier.id ?? -1))
-  }
+  if (!ids) return 0
+
+  const selectedTiers = rewardTiersFromIds({
+    ids,
+    rewardTiers,
+  })
+
   return round(
-    rewardTiers.reduce((subSum, tier) => subSum + tier.contributionFloor, 0),
+    selectedTiers.reduce((subSum, tier) => subSum + tier.contributionFloor, 0),
     6,
```

Comment:
whats the significance of `6`?

Reply:
I think I added this `round` to prevent the long unrounded values which were sometimes appearing in the pay input on NFT selection previously. I thought 6 d.p. was a good number to round to, wdyt?